modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/schema/TableDescriptorImpl.java

```diff
@@ -516,6 +516,11 @@ private Object clone(Object val) throws IgniteCheckedException {
         return fieldName == null ? null : descriptorsMap.get(fieldName);
     }
 
+    /** {@inheritDoc} */
+    @Override public ColumnDescriptor columnDescriptor(int fieldIdx) {
+        return descriptors[fieldIdx];
```

Comment:
What if the given fieldIdx will be outside of the range [descriptors.length)?

Reply:
removed